apps/auxo-app/src/components/Vault/Actions/Withdraw/WithdrawInput.tsx

```diff
@@ -34,39 +37,43 @@ function ApproveWithdrawButton({ withdraw, setWithdraw }: { withdraw: Balance, s
         }
     }, [succeeded, setWithdraw])
 
-    const enterBatchBurn = async () => {
-        try {
-            if (auxoContract) {
-                setApproving(true);
-                const tx = await auxoContract?.enterBatchBurn(withdraw.value)
-                await handleTransaction(tx, eventName, dispatch);
-            } else {
-                console.error('Missing contract details for selected vault')
-            }
-        } catch (err) {
-            dispatch(
-                setAlert({
-                  message: "There was a problem with the transaction",
-                  type: "ERROR",
-                })
-              );
-        } finally {
-            setApproving(false)
+    const newEnterBatchBurn = async () => {
+        if (auxoContract) {
+            setApproving(true);
+            const tx = await auxoContract?.enterBatchBurn(withdraw.value)
+            await handleTx({ tx, dispatch, library, onSuccess: () => {
+                if (vault && vault.userBalances && vault.userBalances.batchBurn) {
+                    const newVault: Vault = {
```

Comment:
Premise: I'm a boomer react dev but It used to be pretty unusual to update the state like this at the component level back to the redux days. The reason is that violates the separation of responsibilities by mixing state management with components in a way that is undesirable. Components are for interaction not state manipulation per se.

Reply:
This makes a lot of sense. I think we can refactor the handle function as a reducer, then simply dispatch the action from the component.

Reply:
This work is now in progress: we are moving state changing to reducers and simply calling "dispatch(setThunkMakeDeposit(args))" in the button, will include in the next commit